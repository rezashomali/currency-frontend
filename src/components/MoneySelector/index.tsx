import React from "react";
import {
  FormControl,
  InputLabel,
  Select,
  MenuItem,
  FormHelperText,
  SelectProps,
} from "@material-ui/core";

type Props = SelectProps & {
  label: string;
  values: {
    [key: string]: string;
  };
  selectedCurrency: string;
};

const MoneySelector: React.FC<Props> = ({
  label,
  values,
  selectedCurrency,
  onChange,
}) => {
  return (
    <FormControl fullWidth>
      <InputLabel shrink>{label}</InputLabel>
      <Select value={selectedCurrency} onChange={onChange}>
        {Object.keys(values).map((item) => (
          <MenuItem key={item} value={item}>
            {values[item]}
          </MenuItem>
        ))}
      </Select>
      <FormHelperText>
        Select the currency you want to conver {label}
      </FormHelperText>
    </FormControl>
  );
};

export default MoneySelector;
