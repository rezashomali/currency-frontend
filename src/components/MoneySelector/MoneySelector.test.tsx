import React from "react";
import renderer from "react-test-renderer";
import MoneySelector from "./index";

test("MoneySelector snapshot ", () => {
  const component = renderer.create(
    <MoneySelector
      label="To"
      values={{
        USD: "$ USD - Dollar",
        EUR: "€ EUR - Euro",
        CHF: "ƒ CHF - Swiss Franc",
      }}
      selectedCurrency="EUR"
      onChange={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
