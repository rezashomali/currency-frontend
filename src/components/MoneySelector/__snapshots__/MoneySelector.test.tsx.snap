// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneySelector snapshot  1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
  >
    To
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      â‚¬ EUR - Euro
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="EUR"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-filled"
  >
    Select the currency you want to conver 
    To
  </p>
</div>
`;
